casepro/msg_board/views.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import unicode_literals
+
+from django.http import JsonResponse, HttpResponse
+from django.core.exceptions import PermissionDenied
+from django.utils import timezone
+
+from smartmin.views import SmartListView, SmartTemplateView
+from smartmin.views import SmartReadView, SmartCRUDL
+from dash.orgs.models import Org
+from dash.orgs.views import OrgPermsMixin, OrgObjPermsMixin
+from casepro.msg_board.models import MessageBoardComment
+
+
+class MessageBoardView(OrgPermsMixin, SmartTemplateView):
+    template_name = 'msg_board/comment_list.haml'
+    permission = 'orgs.org_inbox'
+
+
+class CommentsCRUDL(SmartCRUDL):
+    title = 'Comments'
+    actions = ('list', 'pinned', 'pin', 'unpin')
+    model = MessageBoardComment
+
+    class List(OrgPermsMixin, SmartListView):
+        permission = 'msg_board.messageboardcomment_list'
+        default_order = ('-submit_date',)
+
+        def get_queryset(self):
```

Comment:
Not quite - the intention for pinned comments is that they remained pinned on the side. The main comments stream should still show the context in which the pinned comments were made. Otherwise for people reading the comments stream after the fact, the messages might not make sense without the pinned comments.